### PR TITLE
Add base firmware for the Frog Numpad

### PR DESCRIPTION
**Pull request by ToastyStoemp**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wolf/frog_np/frog_np.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+#include "encoder_action.h"
```
Comment (drashna) [inline keyboards/wolf/frog_np/frog_np.h:18]:
```suggestion
```

File: keyboards/wolf/frog_np/frog_np.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+void matrix_scan_kb(void) {
+    encoder_action_unregister();
+    matrix_scan_user();
+}
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+//    if (!encoder_update_user(index, clockwise)) return false;
+    encoder_action_register(index, clockwise);
+    return true;
+};
```
Comment (drashna) [inline keyboards/wolf/frog_np/frog_np.c:25]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    } else if (index == 1) {
        if (clockwise) {
            tap_code_delay(KC_WH_U, 10);
        } else {
            tap_code_delay(KC_WH_D, 10);
        }
    }

    }
    return true;
}
#endif
```

File: keyboards/wolf/frog_np/frog_np.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+/* Frog NP Keymap Definitions */
+#define LAYOUT( \
+    E00A, E00B, E01A, E01B, \
```
Comment (drashna) [inline keyboards/wolf/frog_np/frog_np.h:22]:
```suggestion
```

File: keyboards/wolf/frog_np/frog_np.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    K00, K01, K02, K03, \
+    K10, K11, K12, K13, \
+    K20, K21, K22, K23, \
+    K30, K31, K32, K33, \
+    K40, K41, K42, K43, \
+    K50, K51, K52, K53  \
+) { \
+    { E00A, E00B, E01A, E01B }, \
```
Comment (drashna) [inline keyboards/wolf/frog_np/frog_np.h:30]:
```suggestion
    { KC_NO, KC_NO, KC_NO, KC_NO }, \
```

File: keyboards/wolf/frog_np/info.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "keyboard_name": "Frog NP", 
+    "url": "", 
+    "maintainer": "ToastyStoemp", 
+    "layouts": {
+        "LAYOUT": {
+            "layout": [{"label":"0,0", "x":2.25, "y":0}, {"label":"0,1", "x":3.25, "y":0}, {"label":"1,0", "x":4.5, "y":0}, {"label":"1,1", "x":5.5, "y":0}, {"label":"1,2", "x":6.5, "y":0}, {"label":"1,3", "x":7.5, "y":0}, {"label":"0,2", "x":8.75, "y":0}, {"label":"0,3", "x":9.75, "y":0}, {"label":"2,0", "x":4.5, "y":1.25}, {"label":"2,1", "x":5.5, "y":1.25}, {"label":"2,2", "x":6.5, "y":1.25}, {"label":"2,3", "x":7.5, "y":1.25}, {"label":"3,0", "x":4.5, "y":2.25}, {"label":"3,1", "x":5.5, "y":2.25}, {"label":"3,2", "x":6.5, "y":2.25}, {"label":"3,3", "x":7.5, "y":2.25}, {"label":"4,0", "x":4.5, "y":3.25}, {"label":"4,1", "x":5.5, "y":3.25}, {"label":"4,2", "x":6.5, "y":3.25}, {"label":"4,3", "x":7.5, "y":3.25}, {"label":"5,0", "x":4.5, "y":4.25}, {"label":"5,1", "x":5.5, "y":4.25}, {"label":"5,2", "x":6.5, "y":4.25}, {"label":"5,3", "x":7.5, "y":4.25}, {"label":"6,0", "x":4.5, "y":5.25}, {"label":"6,1", "x":5.5, "y":5.25}, {"label":"6,2", "x":6.5, "y":5.25}, {"label":"6,3", "x":7.5, "y":5.25}]
```
Comment (drashna) [inline keyboards/wolf/frog_np/info.json:7]:
```suggestion
            "layout": [{{"label":"1,1", "x":5.5, "y":0}, {"label":"1,2", "x":6.5, "y":0}, {"label":"1,3", "x":7.5, "y":0}, {"label":"0,2", "x":8.75, "y":0}, {"label":"0,3", "x":9.75, "y":0}, {"label":"2,0", "x":4.5, "y":1.25}, {"label":"2,1", "x":5.5, "y":1.25}, {"label":"2,2", "x":6.5, "y":1.25}, {"label":"2,3", "x":7.5, "y":1.25}, {"label":"3,0", "x":4.5, "y":2.25}, {"label":"3,1", "x":5.5, "y":2.25}, {"label":"3,2", "x":6.5, "y":2.25}, {"label":"3,3", "x":7.5, "y":2.25}, {"label":"4,0", "x":4.5, "y":3.25}, {"label":"4,1", "x":5.5, "y":3.25}, {"label":"4,2", "x":6.5, "y":3.25}, {"label":"4,3", "x":7.5, "y":3.25}, {"label":"5,0", "x":4.5, "y":4.25}, {"label":"5,1", "x":5.5, "y":4.25}, {"label":"5,2", "x":6.5, "y":4.25}, {"label":"5,3", "x":7.5, "y":4.25}, {"label":"6,0", "x":4.5, "y":5.25}, {"label":"6,1", "x":5.5, "y":5.25}, {"label":"6,2", "x":6.5, "y":5.25}, {"label":"6,3", "x":7.5, "y":5.25}]
```

File: keyboards/wolf/frog_np/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+enum layers {
+    _LAYER0,
+    _LAYER1,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [_LAYER0] = LAYOUT(
+        KC_VOLD, KC_VOLU, KC_WH_D, KC_WH_U,
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/default/keymap.c:25]:
```suggestion
```

File: keyboards/wolf/frog_np/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        KC_F1,   KC_F2,   KC_F3,   KC_F4,
+        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,   KC_P5,   KC_P6,   KC_TRNS,
+        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_P0,   KC_PDOT, KC_PDOT, MO(1)
+    ),
+
+    [_LAYER1] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/default/keymap.c:35]:
```suggestion
```

File: keyboards/wolf/frog_np/readme.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+---
+
+* Keyboard Maintainer: ToastyStoemp
+
+Make example for this keyboard (after setting up your build environment):
+
+    make wolf/Frog NP:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (drashna) [inline keyboards/wolf/frog_np/readme.md:13]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/wolf/frog_np/rules.mk [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = yes       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/wolf/frog_np/rules.mk:20]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/wolf/frog_np/rules.mk [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/wolf/frog_np/rules.mk:23]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by ToastyStoemp** [comment]
Closing until I tested the changes

**Comment by ToastyStoemp** [comment]
Reopened the Pull Request with a rework using the QMK encoder mappings

File: keyboards/wolf/frog_np/frog_np.h [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+/* Frog NP Keymap Definitions */
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/wolf/frog_np/frog_np.h:20]:
```suggestion
#define LAYOUT_orhto_6x4( \
```

File: keyboards/wolf/frog_np/info.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+    "keyboard_name": "Frog NP", 
+    "url": "", 
+    "maintainer": "ToastyStoemp",
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/wolf/frog_np/info.json:6]:
```suggestion
        "LAYOUT_orhto_6x4": {
```

File: keyboards/wolf/frog_np/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+#include QMK_KEYBOARD_H
+
+enum layers {
+    _LAYER0,
+    _LAYER1,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [_LAYER0] = LAYOUT(
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/default/keymap.c:24]:
```suggestion
    [_LAYER0] = LAYOUT_orhto_6x4(
```

File: keyboards/wolf/frog_np/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        KC_F1,   KC_F2,   KC_F3,   KC_F4,
+        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,   KC_P5,   KC_P6,   KC_TRNS,
+        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_P0,   KC_PDOT, KC_PDOT, MO(1)
+    ),
+
+    [_LAYER1] = LAYOUT(
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/default/keymap.c:33]:
```suggestion
    [_LAYER1] = LAYOUT_orhto_6x4(
```

File: keyboards/wolf/frog_np/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+enum layers {
+    _LAYER0,
+    _LAYER1,
+    _LAYER2,
+    _LAYER3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [_LAYER0] = LAYOUT(
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/via/keymap.c:26]:
```suggestion
    [_LAYER0] = LAYOUT_orhto_6x4(
```

File: keyboards/wolf/frog_np/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        KC_F1,   KC_F2,   KC_F3,   KC_F4,
+        KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,   KC_P5,   KC_P6,   KC_TRNS,
+        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_P0,   KC_PDOT, KC_PDOT, MO(1)
+    ),
+
+    [_LAYER1] = LAYOUT(
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/via/keymap.c:35]:
```suggestion
    [_LAYER1] = LAYOUT_orhto_6x4(
```

File: keyboards/wolf/frog_np/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [_LAYER2] = LAYOUT(
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/via/keymap.c:44]:
```suggestion
    [_LAYER2] = LAYOUT_orhto_6x4(
```

File: keyboards/wolf/frog_np/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [_LAYER3] = LAYOUT(
```
Comment (drashna) [inline keyboards/wolf/frog_np/keymaps/via/keymap.c:53]:
```suggestion
    [_LAYER3] = LAYOUT_orhto_6x4(
```

File: keyboards/wolf/frog_np/rules.mk [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes        # Enable encoder
+ENCODER_MAP_ENABLE = yes
```
Comment (drashna) [inline keyboards/wolf/frog_np/rules.mk:23]: Encoder map should only be enabled as the keymap level.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**